src/main/java/org/truffleruby/language/control/RaiseException.java

```diff
@@ -10,17 +10,25 @@
 package org.truffleruby.language.control;
 
 import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.TruffleException;
 import com.oracle.truffle.api.nodes.ControlFlowException;
+import com.oracle.truffle.api.nodes.Node;
 import com.oracle.truffle.api.object.DynamicObject;
 import org.truffleruby.Layouts;
 
-public class RaiseException extends ControlFlowException {
+public class RaiseException extends ControlFlowException implements TruffleException {
 
     private static final long serialVersionUID = -4128190563044417424L;
 
+    private final Node location;
     private final DynamicObject exception;
 
     public RaiseException(DynamicObject exception) {
+        this(null, exception);
+    }
+
+    public RaiseException(Node location, DynamicObject exception) {
+        this.location = location;
```

Comment:
Could we derive the location Node from the backtrace in the exception object? Otherwise it mens we need to pass `this` twice for every exception which becomes quite verbose.